flycheck.el

```diff
@@ -7735,22 +7749,20 @@ contains a cabal file."
     (`haskell-stack-ghc
      (or
       (when (buffer-file-name)
-        (locate-dominating-file (buffer-file-name) "stack.yaml"))
+        (flycheck--locate-dominating-file-matching
+         (file-name-directory
+          (buffer-file-name))
+         "stack.*\\.yaml\\'"))
```

Comment:
Is there some official documentation for the filenames `slack` detects?  I see that `slack-nightly.yaml` is possible, so anything more restrictive than `.*` would probably be too verbose.

Also, what does the `\\'` means here?